hummingbot/connector/exchange/okex/okex_exchange.pyx

```diff
@@ -0,0 +1,1008 @@
+            dict order_books = self._order_book_tracker.order_books
+
+        if trading_pair not in order_books:
+            raise ValueError(f"No order book exists for '{trading_pair}'.")
+        return order_books.get(trading_pair)
+
+    cdef c_did_timeout_tx(self, str tracking_id):
+        self.c_trigger_event(self.MARKET_TRANSACTION_FAILURE_EVENT_TAG,
+                             MarketTransactionFailureEvent(self._current_timestamp, tracking_id))
+
+    cdef c_start_tracking_order(self,
+                                str client_order_id,
+                                str exchange_order_id,
+                                str trading_pair,
+                                object order_type,
+                                object trade_type,
+                                object price,
+                                object amount):
+        self._in_flight_orders[client_order_id] = OkexInFlightOrder(
+            client_order_id=client_order_id,
+            exchange_order_id=exchange_order_id,
+            trading_pair=trading_pair,
+            order_type=order_type,
+            trade_type=trade_type,
+            price=price,
+            amount=amount
+        )
+
+    cdef c_stop_tracking_order(self, str order_id):
+        if order_id in self._in_flight_orders:
+            del self._in_flight_orders[order_id]
+
+    cdef object c_get_order_price_quantum(self, str trading_pair, object price):
+        cdef:
+            TradingRule trading_rule = self._trading_rules[trading_pair]
+        return trading_rule.min_price_increment
+
+    cdef object c_get_order_size_quantum(self, str trading_pair, object order_size):
+        cdef:
+            TradingRule trading_rule = self._trading_rules[trading_pair]
+        return Decimal(trading_rule.min_base_amount_increment)
+
+    cdef object c_quantize_order_amount(self, str trading_pair, object amount, object price=s_decimal_0):
+        cdef:
+            TradingRule trading_rule = self._trading_rules[trading_pair]
+            object quantized_amount = ExchangeBase.c_quantize_order_amount(self, trading_pair, amount)
+            object current_price = self.c_get_price(trading_pair, False)
+            object notional_size
+
+        # Check against min_order_size. If not passing check, return 0.
+        if quantized_amount < trading_rule.min_order_size:
+            return s_decimal_0
+
+        # Check against max_order_size. If not passing check, return maximum.
+        if quantized_amount > trading_rule.max_order_size:
+            return trading_rule.max_order_size
+
+        if price == s_decimal_0:
+            notional_size = current_price * quantized_amount
+        else:
+            notional_size = price * quantized_amount
+        # Add 1% as a safety factor in case the prices changed while making the order.
+
+        if notional_size < trading_rule.min_notional_size * Decimal("1.01"):
+            return s_decimal_0
+
+        return quantized_amount
```

Comment:
There should be def functions for all public cdef functions, e.g. buy(), sell(). See last section of `binance_exchange.pyx` for an example.

Reply:
I did already, I forgot to push the commit.